public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AddAnomalyDetector.tsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import {
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiFlyoutHeader,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiTitle,
+  EuiSpacer,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFormFieldset,
+  EuiCheckableCard,
+} from '@elastic/eui';
+import './styles.scss';
+import CreateNew from './CreateNew';
+
+function AddAnomalyDetector({
+  embeddable,
+  closeFlyout,
+  core,
+  services,
+  mode,
+  setMode,
+  index,
+}) {
+  const onCreate = () => {
+    console.log(`Current mode: ${mode}`);
+    const event = new Event('createDetector');
+    document.dispatchEvent(event);
```

Comment:
What's the purpose of this line? Typically we don't do these low-level JS cmds.

Also, it seems here is where we would want to do event handling (if creation failed, if start detector failed, if saved obj creation failed, etc.